source/dubregistry/api.d

```diff
@@ -145,16 +148,24 @@ override {
 		}
 	}
 
-	Json getInfo(string name) {
-		return m_registry.getPackageInfo(rootOf(name))
+	Json getInfo(string name, bool minimize = false) {
+		return m_registry.getPackageInfo(rootOf(name), false, minimize)
 			.check!(r => r.info.type != Json.Type.null_)(HTTPStatus.notFound, "Package/Version not found")
 			.info;
 	}
 
-	Json getInfo(string name, string ver) {
-		return m_registry.getPackageVersionInfo(rootOf(name), ver)
+	Json getInfo(string name, string ver, bool minimize = false) {
+		return m_registry.getPackageVersionInfo(rootOf(name), ver, minimize)
 			.check!(r => r.type != Json.Type.null_)(HTTPStatus.notFound, "Package/Version not found");
 	}
+
+	Json[string] getPackageInfoWithDependencies(string name, bool minimize = false) {
+		import std.array : assocArray;
+
+		auto aa = m_registry.getPackageInfoWithDependencies(rootOf(name), false, minimize)
+			.check!(r => r !is null)(HTTPStatus.notFound, "Package/Version not found");
+		return aa.byKeyValue.map!((p) => tuple(p.key, p.value.info)).assocArray;
```

Comment:
`assocArray` won't make the GC leaks easier ... it's a shame that streaming Json would require a serious refactoring :/

Reply:
I think the leak has a very defined location, likely a buffer in the mongodb driver.
Collecting those return values should be rather trivial. Also it's a thin copy, the JSON values remain the same.